gnucash/register/register-core/table-allgui.h

```diff
@@ -182,89 +182,32 @@ struct table
 typedef enum
 {
     /* Colors used for background drawing */
-    COLOR_UNKNOWN_BG,
-    COLOR_HEADER_BG,
-    COLOR_PRIMARY_BG,
-    COLOR_PRIMARY_BG_ACTIVE,
-    COLOR_SECONDARY_BG,
-    COLOR_SECONDARY_BG_ACTIVE,
-    COLOR_SPLIT_BG,
-    COLOR_SPLIT_BG_ACTIVE,
+    COLOR_UNKNOWN_BG,          // 0
+    COLOR_HEADER_BG,           // 1
+    COLOR_PRIMARY_BG,          // 2
+    COLOR_PRIMARY_BG_ACTIVE,   // 3
+    COLOR_SECONDARY_BG,        // 4
+    COLOR_SECONDARY_BG_ACTIVE, // 5
+    COLOR_SPLIT_BG,            // 6
+    COLOR_SPLIT_BG_ACTIVE,     // 7
 
     /* Colors used for foreground drawing (text etc)
      * ATTENTION: the background and foreground lists should have
      *            the same types (the same amount of entries) !
      *            The code relies on this ! */
-    COLOR_UNKNOWN_FG,
-    COLOR_HEADER_FG,
-    COLOR_PRIMARY_FG,
-    COLOR_PRIMARY_FG_ACTIVE,
-    COLOR_SECONDARY_FG,
-    COLOR_SECONDARY_FG_ACTIVE,
-    COLOR_SPLIT_FG,
-    COLOR_SPLIT_FG_ACTIVE,
+    COLOR_UNKNOWN_FG,          // 8
+    COLOR_HEADER_FG,           // 9
+    COLOR_PRIMARY_FG,          // 10
+    COLOR_PRIMARY_FG_ACTIVE,   // 11
+    COLOR_SECONDARY_FG,        // 12
+    COLOR_SECONDARY_FG_ACTIVE, // 13
+    COLOR_SPLIT_FG,            // 14
+    COLOR_SPLIT_FG_ACTIVE,     // 15
 
     /* Other colors */
-    COLOR_NEGATIVE, /* Color to use for negative numbers */
+    COLOR_NEGATIVE,            // 16 Color to use for negative numbers
```

Comment:
A first preliminary scan of the uses of the enums suggests the differentiation between FG and BG colors is not used at all in the end. (It is still used to do the calculations, but [the only code that makes style changes based on these results](https://github.com/Bob-IT/gnucash/blob/f1d49047ab754bd938219754e5e9fe07438d0272/gnucash/register/register-gnome/gnucash-sheet.c#L2413) always checks for both FG and BG it one case statement)

So how about redefining the enum as follows:

```
typedef enum {
COLOR_UNDEFINED = 0,
COLOR_HEADER,
COLOR_PRIMARY,
COLOR_PRIMARY_ACTIVE,
COLOR_SECONDARY,
COLOR_SECONDARY_ACTIVE,
COLOR_SPLIT,
COLOR_SPLIT_ACTIVE,
COLOR_NEGATIVE,
} RegisterColor;
```

We can the probably do away with the enum arithmetic and reduce the code even more. But please double-check if my analysis is correct :)

Reply:
So I think this answers my question about keeping foreground and background cell functions

Reply:
So indeed it does answer your question :) Good you had already spotted it.